src/backend/egl/display.rs

```diff
@@ -39,19 +39,53 @@ use tracing::{debug, error, info, info_span, instrument, trace, warn};
 lazy_static::lazy_static! {
     pub(crate) static ref BUFFER_READER: Mutex<Option<WeakBufferReader>> = Mutex::new(None);
 }
+lazy_static::lazy_static! {
+    static ref DISPLAYS: Mutex<HashSet<WeakEGLDisplayHandle>> = Mutex::new(HashSet::new());
+}
 
 /// Wrapper around [`ffi::EGLDisplay`](ffi::egl::types::EGLDisplay) to ensure display is only destroyed
 /// once all resources bound to it have been dropped.
 #[derive(Debug)]
 pub struct EGLDisplayHandle {
     /// ffi EGLDisplay ptr
     pub handle: ffi::egl::types::EGLDisplay,
+    should_terminate: bool,
     _native: Box<dyn std::any::Any + 'static>,
 }
 // EGLDisplay has an internal Mutex
 unsafe impl Send for EGLDisplayHandle {}
 unsafe impl Sync for EGLDisplayHandle {}
 
+#[derive(Clone)]
+struct WeakEGLDisplayHandle {
+    handle: Weak<EGLDisplayHandle>,
+    ptr: u64,
+}
+
+impl From<Arc<EGLDisplayHandle>> for WeakEGLDisplayHandle {
+    fn from(other: Arc<EGLDisplayHandle>) -> Self {
+        let ptr = other.handle as *const c_void as u64;
```

Comment:
Can we keep the original type instead of casting this to u64? imo it should be fine to use `ffi::egl::types::EGLDisplay`  for the ptr in `WeakEGLDisplayHandle` and implement `Send` for it.

Reply:
Good call. Done. Also implemented `Sync` as we do that for the `EGLDisplayHandle` as well and there is no reason why it should be different for the weak variant.